src/test/java/com/fakng/fakngagrgtr/parser/apple/AppleParserTest.java

```diff
@@ -0,0 +1,110 @@
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+@ExtendWith(MockitoExtension.class)
+class AppleParserTest extends AbstractParserTest {
+
+    private static final String URL = "Apple";
+    private AppleParser appleParser;
+    @Mock
+    private WebClient htmlWebClient;
+    @Mock
+    private CompanyRepository companyRepository;
+    @Mock
+    private LocationProcessor locationProcessor;
+
+    @BeforeEach
+    public void init() {
+        appleParser = new AppleParser(
+                htmlWebClient,
+                companyRepository,
+                locationProcessor,
+                URL,
+                new ParserConfig().objectMapper()
+        );
+    }
+
+    @Test
+    void getCompanyName() {
+        assertEquals(appleParser.getCompanyName(), "Apple");
+    }
+
+    @Test
+    public void testParseCreatesVacanciesFromJsonWithCachedLocations() throws IOException {
+        Company company = prepareCompany();
+        Mockito.when(companyRepository.findByTitle(URL))
+                .thenReturn(Optional.of(company));
+        Mockito.when(locationProcessor.processLocation(company, "city_1", "country_1"))
+                .thenReturn(company.getLocations().get(0));
+        Mockito.when(locationProcessor.processLocation(company, "city_2", "country_2"))
+                .thenReturn(company.getLocations().get(1));
+        Mockito.when(htmlWebClient.getPage(URL))
+                .thenReturn(new WebClient().loadHtmlCodeIntoCurrentWindow(Files.readString(Paths.get("src/test/resources/com/fakng/fakngagrgtr/parser/apple-data.html"))));
+
+        appleParser.init();
+
+        List<Vacancy> expectedVacancies = prepareVacancies(company);
+        List<Vacancy> actualVacancies = appleParser.parse();
+
+        assertEquals(expectedVacancies.size(), actualVacancies.size());
+        for (int i = 0; i < expectedVacancies.size(); i++) {
+            assertVacancy(expectedVacancies.get(i), actualVacancies.get(i));
+        }
+    }
+
+    private List<Vacancy> prepareVacancies(Company company) {
+        List<Vacancy> vacancies = new ArrayList<>();
+
+        Vacancy first = new Vacancy();
+        first.setTitle("title_1");
+        first.setUrl("https://jobs.apple.com/en-us/details/apple_1");
+        first.setCompany(company);
+        first.setDescription("team_1\ntitle_1\ndescription_1");
+        first.setLocations(company.getLocations());
+        vacancies.add(first);
+
+        Vacancy second = new Vacancy();
+        second.setTitle("title_2");
+        second.setUrl("https://jobs.apple.com/en-us/details/apple_2");
+        second.setCompany(company);
+        second.setDescription("team_2\ntitle_2\ndescription_2");
+        second.setLocations(company.getLocations().subList(0, 1));
+        vacancies.add(second);
+
+        return vacancies;
+    }
+
+    private void assertVacancy(Vacancy expected, Vacancy actual) {
```

Comment:
Is copied from GoogleParserTest? Please, if so, move this method to the parent class and reuse in both children

Reply:
Well

Reply:
@mighty-mhsl what do you think about adding a few more properties for comparison in assertVacancy (for example, publication date, etc.)?

Reply:
All the fields that we fill in the logic, should be checked in tests. If publishDate is not check, then add it, please. If you see some others - too. But please make sure that ALL tests still pass

Reply:
Okay, I'll add it later. I will update GoogleParser for the new assertVacancy